sycl/include/CL/sycl/detail/spirv.hpp

```diff
@@ -42,15 +42,29 @@ template <typename Group> bool GroupAny(bool pred) {
 }
 
 // Broadcast with scalar local index
+// Work-group supports any integral type
+// Sub-group currently supports only uint32_t
 template <typename Group, typename T, typename IdT>
-detail::enable_if_t<std::is_integral<IdT>::value, T>
+detail::enable_if_t<is_group<Group>::value && std::is_integral<IdT>::value, T>
 GroupBroadcast(T x, IdT local_id) {
   using OCLT = detail::ConvertToOpenCLType_t<T>;
   using OCLIdT = detail::ConvertToOpenCLType_t<IdT>;
   OCLT ocl_x = detail::convertDataToType<T, OCLT>(x);
   OCLIdT ocl_id = detail::convertDataToType<IdT, OCLIdT>(local_id);
   return __spirv_GroupBroadcast(group_scope<Group>::value, ocl_x, ocl_id);
 }
+template <typename Group, typename T, typename IdT>
+detail::enable_if_t<is_sub_group<Group>::value && std::is_integral<IdT>::value,
```

Comment:
If IdT is uint64_t, then it will silently truncated to uint32_t, right?
Would it be useful here to assert or throw an exception if local_id does not fit into 32-bit?

Reply:
Truncation is the desired behavior here.

A sub-group's local ID is represented by an `id<1>` (which is a `size_t`) but the CPU runtime only accepts a SPIR-V `OpGroupBroadcast` instruction with `Subgroup` scope if the `LocalId` parameter is a `uint32_t`.

This version of `GroupBroadcast` can only be called if the `Group` is a sub-group, so truncating into 32 bits should always be safe.